File: Semester 1/Class14_Computer_Contest_Lv_2_Notes_and_Homework_Dec_11/lvl2_hw_sol/lvl2_cell_phone_messaging.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	char key[8][6] = {"abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"};
	string input;
	int pre_row=-1, time=0;
	while (cin >> input && input!="halt")
	{
		pre_row = -1;  time = 0;
		for (int i=0; i<input.size(); i++)
		{
			for (int row=0; row<8; row++)
			{
				for (int col=0; key[row][col]; col++)
				{
					if (input[i] == key[row][col])
					{
						if (row == pre_row)
							time += 2;
						time += col+1;
						pre_row = row;
					}
				}
			}
		}
		cout << time << endl;
	}
    return 0;
}

File: Semester 1/Class14_Computer_Contest_Lv_2_Notes_and_Homework_Dec_11/lvl2_hw_sol/lvl2_cell_phone_messaging.java
import java.util.*;
public class question4 {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		String [] key = {"", "", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"};
		while(true){
			String msg = in.nextLine();  int sum = 0, prev = 0;
			if(msg.equals("halt")) break;
			for(int i=0; i<msg.length(); i++){
				String c = ""+msg.charAt(i);
				for(int k=2; k<=9; k++){
					if(key[k].contains(c)){
						int t = key[k].indexOf(c)+1;
						sum += t;
						if(k == prev) sum += 2; //add pause
						prev = k;
					}
				}
			}
			System.out.println(sum);
		}
	}
}